src/components/DropzoneDialogFunc.js

```diff
@@ -0,0 +1,156 @@
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import PropTypes from "prop-types";
+import React, { useState, useEffect } from "react";
+import { Fragment } from "react";
+
+import DropzoneAreaFunc from "./DropzoneAreaFunc";
+
+/**
+ * This component provides the DropzoneArea inside of a Material-UI Dialog.
+ *
+ * It supports all the Props and Methods from `DropzoneArea`.
+ */
+const DropzoneDialogFunc = (props) => {
+  const {
+    cancelButtonText,
+    dialogProps,
+    dialogTitle,
+    fullWidth,
+    maxWidth,
+    onChange,
+    onClose,
+    onSave,
+    open,
+    submitButtonText,
+    ...dropzoneAreaProps
+  } = props;
+
+  const [currentFiles, setCurrentFiles] = useState([]);
+  const [submitDisabled, setSubmitDisabled] = useState(true);
```

Comment:
There's no need to keep a `submitDisabled` state, computing the boolean on each render based on `currentFiles.length === 0` is perfectly fine.